docs/learn-spree.md

```diff
@@ -4,36 +4,61 @@ title: SPREE
+will be executed, i.e. how the receiving parachain will interpret the message. While ICMP accomplishes trustless message
+passing, SPREE is the trustless interpration of the message and a key part to the usefulness of ICMP.
 
-The description was later simplified in a presentation at Tokyo DOT Day:
+SPREE modules are like recipes in cookbooks. For example, if we give an order to a cook to make a souffle, and we’re
+decently confident in the ability of the cook, we have a vague idea of what will be made but no actually surety how it 
+will be made. However, let’s say that a cook has the “Souffle Maker’s Manual” on their bookshelf and has committed
+themselves to only make souffles from this book. Now we can also consult the same book that the cook has, and we have a
+precise understanding of what will happen when we tell the cook to make a souffle. In this example, “make a souffle” was
+the message in ICMP and the cookbook was the SPREE module.
 
-- Parachains can opt-in to special runtime logic fragments (like smart contracts)
-- Own storage, own ICMP endpoint
-- All instances across parachains have identical logic
-- Executes alongside parachain
-- Protected: storage can not be altered by parachain logic; messages can not be faked from them by parachains
+In concrete terms, SPREE modules could be useful for various functionally on Polkadot. One suggested use case of SPREE
+modules is for a trustless decentralized exchange that is offered as functionality to any parachain without any extra
+effort from parachain developers. One can imagine this working by having a SPREE module which exposes the interface for
+the incrementing and decrementing of balances of various assets based on a unique identifier.
 
 ## Why?
 
-Sending messages across parachains in ICMP only ensures that the message will be delivered but does not specify the code that will be executed, or how the message will be interpreted by the receiving parachain. There would be ways around this such as requesting a verifiable receipt of the execution from the receiving parachain, but in the naked case the other parachain would have to be trusted. Having shared code which exist in appendices that the parachain can opt-in to supporting resolves the need for trust and makes the execution of the appendices completely trustless.
+Sending messages across parachains in ICMP only ensures that the message will be delivered but does not specify the code
+that will be executed, or how the message will be interpreted by the receiving parachain. There would be ways around
+this such as requesting a verifiable receipt of the execution from the receiving parachain, but in the naked case the
+other parachain would have to be trusted. Having shared code which exist in appendices that the parachain can opt-in to
+supporting resolves the need for trust and makes the execution of the appendices completely trustless.
```

Comment:
```suggestion
resolves the need for trust and makes the execution of the appendices completely trustless.
```

Reply:
Resolves the need for trust + makes ... completely trustless = unnecessary repetition?